markdown/developers/tutorials/dsl2_modules_tutorial.md

```diff
@@ -11,414 +11,298 @@ If you create a new module with the goal of contributing the code to *nf-core*,
-
-Unfortunately, FGBIO manages to cause an error exit even with this solution, and we are therefore forced to output the version as a string, like this
+### Test data
 
-```nextflow
-def VERSION = '1.3.0'
-```
+Even before beginning the development of a module, you should identify a small dataset you can use to test its functionality. Ideally, the dataset is existing already and can be shared with other test workflows for other modules.
 
-and then in the script
+> :recycle: This is in active development, keep an eye for available test data [here](https://github.com/nf-core/test-datasets/tree/modules/data) and how to access them using a config file (see this [change](https://github.com/nf-core/modules/pull/365)).
 
-```bash
-echo $VERSION >${software}.version.txt
-```
+---
 
-Our final script will therefore look like this:
+## Writing a new module
 
-```nextflow
-// Import generic module functions
-include { initOptions; saveFiles; getSoftwareName } from './functions'
+Check out [Tutorial: Create a DSL2 Module](https://nf-co.re/developers/tutorials/dsl2_modules_tutorial) for how to create a new DSL2 Module
 
-params.options = [:]
-options        = initOptions(params.options)
+### New module workflow
 
-def VERSION = '1.3.0'
+We have implemented a number of commands in the `nf-core/tools` package to make it incredibly easy for you to create and contribute your own modules to nf-core/modules.
 
-/*
-unfortunately need to output the version manually
-as done for module
-https://github.com/nf-core/modules/blob/master/software/homer/annotatepeaks/main.nf
-because the solution adopted in iVar, i.e.
-echo \$(fgbio --version 2>&1) >${software}.version.txt
-for FGBIO still generates an error exit in Nextflow for some reasons
-*/
+1. Install the latest version of [`nf-core/tools`](https://github.com/nf-core/tools#installation) (`>=2.1`)
 
-process FGBIO_FASTQTOBAM {
-    tag "$meta.id"
-    label 'process_low'
-    publishDir "${params.outdir}",
-        mode: params.publish_dir_mode,
-        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+    > ⚠️ [2021-11-26] please use `dev` version of tools
 
-    conda (params.enable_conda ? "bioconda::fgbio=1.3.0" : null)
-    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
-        container "https://depot.galaxyproject.org/singularity/fgbio:1.3.0--0"
-    } else {
-        container "quay.io/biocontainers/fgbio:1.3.0--0"
-    }
+2. Install [`Nextflow`](https://www.nextflow.io/docs/latest/getstarted.html#installation) (`>=21.04.0`)
+3. Install any of [`Docker`](https://docs.docker.com/engine/installation/), [`Singularity`](https://www.sylabs.io/guides/3.0/user-guide/) or [`Conda`](https://conda.io/miniconda.html)
+4. [Fork and clone the nf-core/modules repo locally](#uploading-to-nf-coremodules)
```

Comment:
I understand this is meant to make the checklist leaner, but a tutorial in my view should be a step-by-step. and if you fork before start coding, people should get the instructions about forking before the instructions about the code.
in this way, they're linked to content that's placed after the guidelines on testing.
the order of the description therefore doesn't follow the order of actions that should be taken.